app/chat/_api/useGetChats.ts

```diff
@@ -1,15 +1,24 @@
-import { useQuery } from "@tanstack/react-query";
+import { useInfiniteQuery } from "@tanstack/react-query";
 
+import { ResponseData } from "@/app/_common/types/response";
+
+import { ChatType } from "../_types/chat";
 import { getChats } from "./index";
 
 const useGetChats = () => {
-  const { isLoading, data } = useQuery({
+  return useInfiniteQuery<ResponseData<ChatType> | undefined>({
     queryKey: ["chats"] as const,
-    queryFn: getChats,
+    queryFn: context => {
+      const cursorId = context.pageParam as ChatType;
+      return getChats(cursorId.cursorId);
```

Comment:
구조 분해 할당을 사용해보는 것도 좋을 것 같아요.

```suggestion
      const { cursorId } = context.pageParam as ChatType;
      return getChats(cursorId);
```